fs/fs_zenfs.cc

```diff
@@ -441,9 +441,9 @@ IOStatus ZenFS::SyncFileMetadata(ZoneFile* zoneFile, bool replace) {
 
   std::lock_guard<std::mutex> lock(files_mtx_);
 
-  if (GetFileInternal(zoneFile->GetFilename()) == nullptr) {
-    Info(logger_, "File %s doesn't exist, skip sync file metadata!",
-         zoneFile->GetFilename().data());
+  if (zoneFile->IsDeleted()) {
```

Comment:
If we are holding the files_mtx_, and the `GetFileInternal` returns true, then this file surely exists. But if we simply check `is_deleted_` flag here, it may return false (means the file exists), but the file actually had been deleted by other threads, in this case, we may migrate a deleted file.

Reply:
Oh, never mind, I just realized the `DeleteFile` is also holding file_mtx_